transformers/run_eval.py

```diff
@@ -4,47 +4,95 @@
 from transformers import pipeline
 import evaluate
 from normalizer import data_utils
+import time
+from tqdm import tqdm
 
 wer_metric = evaluate.load("wer")
 
 
-def dataset_iterator(dataset):
-    for i, item in enumerate(dataset):
-        yield {**item["audio"], "reference": item["norm_text"]}
-
-
 def main(args):
     asr_pipe = pipeline(
-        "automatic-speech-recognition", model=args.model_id, device=args.device
+        "automatic-speech-recognition", model=args.model_id, device=args.device, batch_size=args.batch_size
```

Comment:
Note that RTFX will be sensitive to batch size

Reply:
Very good point, I think we discussed this when we pushed initial changes.
For WER, we used bs=8
and for RTFX, we used bs=1

Reply:
For 1. Some of the models run into OOM issues with a larger batch size and to keep the ground same for all models, reduced bs to 8.

Reply:
If you're doing offline transcription, you're typically interested in running as fast as possible, and increasing batch size until you fill up your GPU's memory is a good first attempt at doing that (this does not always work because some inputs may be take much longer to process with something like RNN-T or Whisper). For streaming inference you don't want your batch size too large, but I think we can ignore that for now.

> Some of the models run into OOM issues with a larger batch size and to keep the ground same for all models, reduced bs to 8.

Sure, but why should other models be held back just because one model can't scale beyond batch size 8? For perspective, I ran at batch size 200 on A100-80GB in https://arxiv.org/pdf/2311.04996.pdf using NeMo models. It is possible to achieve 4000 RTFx with greed decoding for stt_en_conformer_ctc_large, but the current leaderboard gets only 133 RTFx using this batch size 1 constraint. So I hope that illustrates the issue with artificially limiting batch size.

I would say "use the batch size that gives the best perf" (which normally is a larger one) is a good choice. Leave it to the person responsible for the submission what the best batch size is.